azure-iot-hub-devicesdk/samples/send_to_output_module.py

```diff
@@ -0,0 +1,50 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import os
+import time
+from azure.iot.hub.devicesdk.module_client import ModuleClient
+from azure.iot.hub.devicesdk.auth.authentication_provider_factory import from_connection_string
+from azure.iot.hub.devicesdk.message import Message
+import uuid
+
+# The connection string for a device should never be stored in code. For the sake of simplicity we're using an environment variable here.
+conn_str = os.getenv("IOTHUB_MODULE_CONNECTION_STRING")
+# The "Authentication Provider" is the object in charge of creating authentication "tokens" for the device client.
+auth_provider = from_connection_string(conn_str)
+# For now, the SDK only supports MQTT as a protocol. the client object is used to interact with your Azure IoT hub.
+# It needs an Authentication Provider to secure the communication with the hub, using either tokens or x509 certificates
+module_client = ModuleClient.from_authentication_provider(auth_provider, "mqtt")
+
+
+# The connection state callback allows us to detect when the client is connected and disconnected:
+def connection_state_callback(status):
+    print("connection status: " + status)
+
+
+# Register the connection state callback with the client...
+module_client.on_connection_state = connection_state_callback
+
+# ... and connect the client.
+module_client.connect()
+
+# send 5 messages with a 1 second pause between each message
+for i in range(0, 5):
+    print("sending message #" + str(i))
+    msg = Message("test wind speed " + str(i))
+    msg.message_id = uuid.uuid4()
+    msg.correlation_id = "correlation-1234"
+    msg.custom_properties["tornado-warning"] = "yes"
+    module_client.send_to_output(msg, "twister")
+    time.sleep(1)
```

Comment:
this block confuses the sample.  If you want to show how these properties work, maybe we should have a property sample.

Reply:
are you referring to the properties set on the message ? I think in the past I have changed the device client sample too to have some properties.  are you referring to the fact that we should just send only string messages ? 

---
In reply to: [246254489](https://github.com/Azure/azure-iot-sdk-python-preview/pull/35#discussion_r246254489) [](ancestors = 246254489)